dpypelines/pipeline/dataset_ingress_v1.py

```diff
@@ -345,94 +349,127 @@ def dataset_ingress_v1(files_dir: str, pipeline_config: dict):
 
     # TODO - validate the csv once we know what we're validating
 
-    # Upload output files to Upload Service
-    try:
-        # Create UploadClient from upload_url
-        upload_client = UploadServiceClient(upload_url)
-        logger.info(
-            "UploadClient created from upload_url", data={"upload_url": upload_url}
-        )
-    except Exception as err:
-        logger.error(
-            "Error creating UploadClient", err, data={"upload_url": upload_url}
-        )
-        de_notifier.failure()
-        raise err
+    # Allow DE's to skip the upload to s3 part of the pipeline while
+    # developing code locally.
+    skip_data_upload = os.environ.get("SKIP_DATA_UPLOAD", False)
+    if skip_data_upload is not False:
+        if skip_data_upload.strip() in ["True", "true"]:
+            skip_data_upload = True
+        elif skip_data_upload.strip() in ["False", "false"]:
+            skip_data_upload = False
+        else:
+            # Catch a trivial raise as we need the stack trace of the
+            # error for the logger, so it needs to be a raised error.
+            try:
+                raise ValueError(
+                    "If present SKIP_DATA_UPLOAD must be one of: True, true, False, false."
+                )
+            except ValueError as err:
```

Comment:
Should this be `ValueError` or `Exception`?

Reply:
ValueError should be fine, its nice and explicit. Lets keep it handling just the scenario we're creating.